bundles/org.eclipse.emfcloud.modelserver.emf/src/org/eclipse/emfcloud/modelserver/emf/common/DefaultModelController.java

```diff
@@ -72,7 +72,10 @@ public DefaultModelController(final ModelRepository modelRepository, final Sessi
    public void create(final Context ctx, final String modeluri) {
       Optional<EObject> root = readPayload(ctx);
       if (root.isEmpty()) {
-         badRequest(ctx, "Create new model failed.");
+         return;
+      }
+      if (this.modelRepository.getModel(modeluri).isPresent()) {
+         conflict(ctx, "Model already exists.");
```

Comment:
This check seems to break the createModel request.
As the model is about to being created, `getModel` return an empty optional in that case.

Reply:
You are absolutely right, thank you! It seems that `getModel` already tries to load the model, so I changed the query to the safer `hasModel`